src/main/java/seedu/storemando/ui/TablePanel.java

```diff
@@ -0,0 +1,108 @@
+package seedu.storemando.ui;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+import java.net.URL;
+import java.time.LocalDate;
+import java.util.ResourceBundle;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.scene.layout.Region;
+import seedu.storemando.model.item.Item;
+
+/**
+ * Table displaying all items expiring in a week
+ */
+public class TablePanel extends UiPart<Region> implements Initializable {
+    private static final String FXML = "TablePanel.fxml";
+    private ObservableList<Item> itemList;
+
+    @FXML
+    private TableView<ExpiringItem> tableView;
+
+    @FXML
+    private TableColumn<ExpiringItem, String> daysBeforeExpiryCol;
+
+    @FXML
+    private TableColumn<ExpiringItem, String> numberOfItems;
+
+    /**
+     * Creates a {@code TablePanel} with the given {@code ObservableList}.
+     */
+    public TablePanel(ObservableList<Item> itemList) {
+        super(FXML);
+        this.itemList = itemList;
+        tableView.setItems(getExpiringItems(itemList));
+    }
+
+    /**
+     * Passing the data into the table.
+     * @param itemList The list of data required by the table.
+     * @return A list of items expiring in a week.
+     */
+    private ObservableList<ExpiringItem> getExpiringItems(ObservableList<Item> itemList) {
+        ObservableList<ExpiringItem> tableRows = FXCollections.observableArrayList();
+        for (int i = 1; i <= 7; i++) {
+            tableRows.add(new ExpiringItem(i, itemList));
```

Comment:
Can you clarify if this would break in the event of an itemList of size smaller than 7 being added in?

Reply:
Nope it wouldn't. The value 1 to 7 signifies the number of days from today. It's not iterating through the items.